activerecord/test/cases/relation/where_chain_test.rb

```diff
@@ -121,8 +121,8 @@ def test_rewhere_with_alias_condition
     end
 
     def test_rewhere_with_nested_condition
-      relation = Post.where.missing(:comments).rewhere("comments.id": comments(:does_it_hurt))
-      expected = Post.left_joins(:comments).where("comments.id": comments(:does_it_hurt))
+      relation = Post.where.missing(:comments).rewhere("comments.post_id": posts(:thinking))
+      expected = Post.left_joins(:comments).where("comments.post_id": posts(:thinking))
```

Comment:
Is it necessary to change from #id to #post_id? This seems like overreaching the scope of the PR which is just to update for postgres only

Reply:
Yes it is necessary, the test was check `rewhere` would be able to override `missing` and the PR changes `missing` to use #post_id hence this change.

Reply:
I guess my concern is that this is a test that applies to all adapters not just to postgres. Perhaps it would be optimal to find a solution that only affects postgres?